appcelerator/titanium_mobile
# fix(android): accelerometer logs warnings after app exit

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28272

**Summary:**
- After setting up an "update" event listener with `Ti.Accelerometer`, a lot of warnings will be logged after app exit if you don't remove the listener.
- Will now automatically unregister the sensor listener when JS runtime is terminated.

**Test:**
1. Build and run the below code on a real/physical device.
2. Verify the following messages are logged when you move the device.
`[INFO]  @@@ (x,y,z) = (<Number>,<Number>,<Number>)`
3. Back out of the app.
4. Verify that the above messages stop logging. _(Might take a couple seconds.)_
5. Verify that the below warning is NOT logged. _(This is the fix.)_
`[WARN] V8Object: (main) [xx,xx] Runtime disposed, cannot fire event 'update'`
6. Re-launch the app.
7. Verify that accelerometer data is being logged again.

```javascript
Ti.Accelerometer.addEventListener("update", (e) => {
	Ti.API.info(`@@@ (x,y,z) = (${e.x},${e.y},${e.z})`);
});
```
